mirgecom/steppers.py

```diff
@@ -164,41 +188,60 @@ def _advance_state_leap(rhs, timestepper, get_timestep,
     if t_final <= t:
         return istep, t, state
 
-    # Generate code for Leap method.
-    dt = get_timestep(state=state)
-    stepper_cls = generate_singlerate_leap_advancer(timestepper, component_id,
-                                                    rhs, t, dt, state)
-    while t < t_final:
-
-        if logmgr:
-            logmgr.tick_before()
-
+    try:
+        # Generate code for Leap method.
         dt = get_timestep(state=state)
-        if dt < 0:
-            return istep, t, state
+        stepper_cls = generate_singlerate_leap_advancer(timestepper, component_id,
+                                                        rhs, t, dt, state)
+    except SynchronizedException as synced_exception:
+        if exception_callback is not None:
+            state, stop = exception_callback(istep, t, state,
+                synced_exception.exception)
+            if stop:
+                return istep, t, state
```

Comment:
Why is this not `break` like here: https://github.com/illinois-ceesd/mirgecom/pull/376/files#diff-e13bce10f3dc4e997e0c56c91b3d0309d0ebf5ceaecd41ae16bca424c91ff24fR125?

Reply:
The diff makes it hard to see, but this section is actually outside of the timestepping loop.